core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -223,7 +223,13 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    *         not exist
    */
   def deleteIfExists(): Boolean = {
-    closeHandler()
+    inLock(lock) {
```

Comment:
Changes in this file seem completely irrelevant to this PR, unless an explanation is provided. Before diving more into this, is it intended to bring in these changes here? and if so, why? This also grants a comment in the commit log if these changes here are legit.

Reply:
Thanks for the catch, this was a sloppy rebase!